src/libsync/discovery.cpp

```diff
@@ -1134,6 +1170,16 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
             item->_status = SyncFileItem::Status::NormalError;
         }
 
+        if (const auto folderQuota = folderQuotaAvailable(item);item->_size > folderQuota && folderQuota > -1) {
+            item->_instruction = CSYNC_INSTRUCTION_ERROR;
+            const auto folderName =
+            item->_errorString = tr("Upload of %1 exceeds the %2 of space left in %3 folder.", "folder sizes, folder name").arg(Utility::octetsToString(item->_size),
+                                                                                                Utility::octetsToString(folderQuota),
+                                                                                                _currentFolder._server.isEmpty() ? "root" : _currentFolder._server);
+            item->_status = SyncFileItem::Status::NormalError;
+            _discoveryData->_anotherSyncNeeded = true;
```

Comment:
why ?
would not this make it loop another synchronization forever ?
should we not also add it to a temporary blacklist to delay next retry ?